src/condor_dagman/parse.cpp

```diff
@@ -1745,8 +1745,27 @@ static bool parse_vars(Dag *dag, const char *filename, int lineNumber)
 
 	std::string varName;
 	std::string varValue;
-
-	char *varsStr = strtok( NULL, "\n" ); // just get all the rest -- we'll be doing this by hand
+	bool prepend; //Bool for if variable is prepended or appended. Default is false
+
+	char *varsStr = strtok( NULL, "\n" ); // just get all the rest and we will manually parse vars
+
+/*	
+*	Check to see if PREPEND or APPEND was specified before variables to be passed.
+*	If option is found then we set prepend boolean appropriately and increment
+*	the original varsStr pointer to point at char after option. -Cole Bollig
+*/
+	std::string temp(varsStr);//Make string with current token
+        if ( temp.find("PREPEND") != std::string::npos ){//See if PREPEND is a sub-string
+		prepend = true;
+		varsStr += 7;
+	} else if ( temp.find("APPEND") != std::string::npos ){//See if APPEND is a sub-string
+		prepend = false;
+		varsStr += 6;
```

Comment:
temp.find("BOY")  will "find"  "BOY" when the value is "POOLBOY".  you should use a more exact check for the keywords you expect.